libs/core/algorithms/include/hpx/parallel/algorithms/detail/find.hpp

```diff
@@ -256,4 +256,221 @@ namespace hpx { namespace parallel { inline namespace v1 { namespace detail {
             part_count, tok, std::forward<F>(f), std::forward<Proj>(proj));
     }
 #endif
+
+    // provide implementation of std::find_end supporting iterators/sentinels
+    template <typename Iter1, typename Sent1, typename Iter2, typename Sent2,
+        typename Pred, typename Proj1, typename Proj2>
+    constexpr Iter1 sequential_search(Iter1 first1, Sent1 last1, Iter2 first2,
+        Sent2 last2, Pred&& op, Proj1&& proj1, Proj2&& proj2)
+    {
+        for (/**/; /**/; ++first1)
+        {
+            Iter1 it1 = first1;
+            for (Iter2 it2 = first2; /**/; (void) ++it1, ++it2)
+            {
+                if (it2 == last2)
+                {
+                    return first1;
+                }
+                if (it1 == last1)
+                {
+                    return last1;
+                }
+                if (!hpx::util::invoke(op, hpx::util::invoke(proj1, *it1),
+                        hpx::util::invoke(proj2, *it2)))
+                {
+                    break;
+                }
+            }
+        }
+    }
+
+    template <typename ExPolicy>
+    struct sequential_find_end_t
+      : hpx::functional::detail::tag_fallback<sequential_find_end_t<ExPolicy>>
+    {
+    private:
+        template <typename Iter1, typename Sent1, typename Iter2,
+            typename Sent2, typename Pred, typename Proj1, typename Proj2>
+        friend inline constexpr Iter1 tag_fallback_invoke(
+            sequential_find_end_t<ExPolicy>, Iter1 first1, Sent1 last1,
+            Iter2 first2, Sent2 last2, Pred&& op, Proj1&& proj1, Proj2&& proj2)
+        {
+            if (first2 == last2)
+            {
+                return detail::advance_to_sentinel(first1, last1);
+            }
+
+            Iter1 result = last1;
+            while (true)
+            {
+                Iter1 new_result = sequential_search(
+                    first1, last1, first2, last2, op, proj1, proj2);
+
+                if (new_result == last1)
+                {
+                    break;
+                }
+                else
+                {
+                    result = new_result;
+                    first1 = result;
+                    ++first1;
+                }
+            }
+            return result;
+        }
+
+        template <typename Iter1, typename Iter2, typename Token, typename Pred,
+            typename Proj1, typename Proj2>
+        friend inline constexpr void tag_fallback_invoke(
+            sequential_find_end_t<ExPolicy>, Iter1 it, Iter2 first2,
+            std::size_t base_idx, std::size_t part_size, std::size_t diff,
+            Token& tok, Pred&& op, Proj1&& proj1, Proj2&& proj2)
+        {
+            util::loop_idx_n<ExPolicy>(base_idx, it, part_size, tok,
+                [=, &tok, &op, &proj1, &proj2](auto t, std::size_t i) -> void {
+                    if (hpx::util::invoke(op, hpx::util::invoke(proj1, t),
+                            hpx::util::invoke(proj2, *first2)))
```

Comment:
Please use `HPX_INVOKE()` instead of `hpx::util::invoke()` in implementation files (not tests, etc.)